pkg/network/passt/manager.go

```diff
@@ -0,0 +1,121 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors.
+ *
+ */
+
+package passt
+
+import (
+	"context"
+
+	"path/filepath"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/log"
+
+	netvmispec "kubevirt.io/kubevirt/pkg/network/vmispec"
+)
+
+type command interface {
+	Start() error
+	Wait() error
+	Kill() error
+	String() string
+	Stderr() string
+}
+
+type clusterConfigurer interface {
+	GetNetworkBindings() map[string]v1.InterfaceBindingPlugin
+}
+
+type RepairLifecycleManager struct {
+	clusterConfigurer clusterConfigurer
+	mutex             sync.Mutex
+	activeRunners     map[types.UID]struct{}
+}
+
+func NewRepairLifecycleManager(clusterConfigurer clusterConfigurer) *RepairLifecycleManager {
+	return &RepairLifecycleManager{
+		clusterConfigurer: clusterConfigurer,
+		activeRunners:     make(map[types.UID]struct{}),
+	}
+}
+
+func (s *RepairLifecycleManager) Handle(vmi *v1.VirtualMachineInstance,
+	discoverFunc func(*v1.VirtualMachineInstance) (string, error)) error {
+
+	bindings := s.clusterConfigurer.GetNetworkBindings()
+	binding := netvmispec.GetPodNetworkInterfaceBinding(vmi, bindings)
+	if binding == nil || binding.DomainAttachmentType == v1.ManagedTap || binding.DomainAttachmentType == v1.Tap {
+		return nil
+	}
+
+	passtSocketDir, err := discoverFunc(vmi)
+	if err != nil {
+		return err
+	}
+
+	symlinkToDir, err := CreateShortenedSymlink(passtSocketDir, string(filepath.Separator))
+	if err != nil {
+		return err
+	}
+
+	repairSocketPath, err := FindRepairSocketInDir(symlinkToDir)
+	if err != nil {
+		return err
+	}
+
+	//Socket not found
+	if repairSocketPath == "" {
+		repairSocketPath = symlinkToDir
+	}
```

Comment:
since the fallback is the directory, why not always just use the directory, if the passt-repair knows to find the socket anyhow ?
(unless i miss something)

Reply:
so this is raceful, we can miss the socket and then it is created and we call with fallback isnt it ?

Reply:
i already wrote above that it isnt good fwiw